src/extensions/speech_extension/commands/transcribe_command.cs

```diff
@@ -0,0 +1,323 @@
+using System.Text;
+using System.Text.Json;
+using System.Net;
+using System.Net.Http.Headers;
+
+namespace Azure.AI.Details.Common.CLI
+{
+    public class TranscribeCommand : Command
+    {
+        public TranscribeCommand(ICommandValues values) : base(values)
+        {
+            _quiet = _values.GetOrDefault("x.quiet", false);
+            _verbose = _values.GetOrDefault("x.verbose", false);
+        }
+
+        public bool RunCommand()
+        {
+            try
+            {
+                Transcribe();
+            }
+            catch (WebException ex)
+            {
+                FileHelpers.LogException(_values, ex);
+                ConsoleHelpers.WriteLineError($"\n  ERROR: {ex.Message}");
+                JsonHelpers.PrintJson(HttpHelpers.ReadWriteJson(ex.Response, _values, "transcribe"));
+            }
+
+            return _values.GetOrDefault("passed", true);
+        }
+
+        private void Transcribe()
+        {
+            StartCommand();
+
+            var content = new MultipartFormDataContent();
+            content.Add(new ByteArrayContent(FileHelpers.ReadAllBytes(GetAudioFile())), "audio", Path.GetFileName(GetAudioFile()));
```

Comment:
>new ByteArrayContent(FileHelpers.ReadAllBytes(GetAudioFile()) [](http://example.com/codeflow?start=24&length=61)

Rather than reading the file into memory and then copying it into a byte array, wouldn't it be better to use a StreamContent, pass the file as a stream into it and let dotnet handle streaming?

Reply:
I haven't ever used "StreamContent" before... i'm using the `FileHelpers` methods cause they deal with things like finding the file on the "data" path, if you've also passed in something like `--input-path FOOBAR`... 

Since the service has a limit on file size alraedy, andit's lower than the memory that somebody'd have on the device they're calling this from (bold statement), i think it's fine. I suspect you're right though, that `StreamContent` would be better overall. maybe log an issue on that? We can see if the bot can fix that one independently if you want. :-)

Reply:
FileHelpers.GetStream() would marry both benefits sounds like....

Reply:
updated... (will be in next push)